internal/cluster/cluster.go

```diff
@@ -86,14 +87,17 @@ func (c *clusterAPI) BuildAndSign(
 	mcm hubv1beta1.ManagedClusterModule,
 	cluster clusterv1.ManagedCluster) (bool, error) {
 
-	requeue := false
-
 	modSpec := mcm.Spec.ModuleSpec
 	mappings, err := c.kernelMappingsByKernelVersion(ctx, &modSpec, cluster)
 	if err != nil {
 		return false, err
 	}
 
+	// if no mappings were found, return not completed
+	if len(mappings) == 0 {
+		return false, nil
```

Comment:
1. At this point we assume that the mapping contains everything needed already?
2. Why don't we return `true` here? If we have nothing to do then we are done aren't we? (same logic that you wrote for `handleSigning`)

Reply:
1. Yes, we assume that all the needed data is stored in the KernelMapping
2. If no mappings are found, we don't want to continue to create ManifestWork, since we don't have suitable images to deploy

Reply:
Makes sense.